application/common/models/Reset.php

```diff
@@ -337,4 +377,156 @@ public static function getExpireTimestamp()
 
         return $time + $params['reset']['lifetimeSeconds'];
     }
+
+    /**
+     * Check if this reset is currently disabled
+     * @return bool
+     */
+    public function isDisabled()
+    {
+        if ($this->disable_until !== null) {
+            $disableUntilTime = strtotime($this->disable_until);
+            // Intentionally loose comparison to catch zero
+            if ($disableUntilTime == false) {
+                return true;
+            }
+            return $disableUntilTime > time();
```

Comment:
We generate the datetime entry from php and then insert it, so we don't use the db to generate dates/times at all.